### DISCUSSION TITLE
Using CPX for UART connection between Luckfox board and Crazyflie drone

### DISCUSSION THREAD

**Discussion by Pitro979** [bitcraze/discussions#1567] [Q&A]
Hello,
I would like to get a similar setup to [this tutorial](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/development/uart_communication/) working using the following hardware:

    Luckfox Pico Mini B
    Crazyflie 2.1 (not BOLT)

The product page for the board can be found [here](https://www.luckfox.com/Luckfox-Pico-Mini-B), and in the wiki [here]( https://wiki.luckfox.com/Luckfox-Pico/Luckfox-Pico-UART).

With the goal of using the Luckfox board (onboard the drone since it's tiny) as the "client" for the drone, as if it was a PC connecting over Crazyradio, using serial communication instead of CR.
We would use one of the expansion connectors on the CF to connect to a UART interface on the Luckfox, for instance pins 10 and 11 for UART3.
We have some doubts regarding power delivery to the board which can be either handled via USB, or the VBUS and ground lines on pins 1 and 2, which would need a 5V and ground rail respectively.

In particular we have the following questions:

    Which is the best way to power the Luckfox board onboard the CF?
    Would the prototyping deck or expansion deck be needed to access relevant CF pins?
    What would be needed (as software) on the CF side to send commands and receive telemetry data between Luckfox and CF over UART?

**Comment by knmcguire** [comment]
Hi!

* I can't find the power requirements for this board, but I would guess that you could try Vcom port for starters if it has it's own power management (but make sure you know what your are doing): https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#expansion-port-pinout
* Yes prototype board could work, but you can also use a battery deck for this. I made this makeshift deck with the xiao sense board from Seeed. 

![20240924_105147](https://github.com/user-attachments/assets/b5252bba-6b8b-4e60-9b7e-67850530e802)

* I recommend you to take a look at this documentation: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/development/uart_communication/. As long as you have the tx/rx/gnd connected then it should work. You can also take a look at an even simplere example right here: https://github.com/knmcguire/crazyflie-firmware/tree/bolt-pos-through-uart/examples/app_uart
